InfrastructureBackupValidator/BackupValidationTool.psm1

```diff
@@ -0,0 +1,591 @@
+        [SecureString]
+        $DecryptionCertPassword,
+
+        [Parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $SQLServerInstanceName = "localhost\SQLEXPRESS", # Use default SQL Express server instance
+
+        [Parameter(Mandatory = $false)]
+        [ValidateNotNullOrEmpty()]
+        [PSCredential]
+        $SQLCredential,
+
+        [Parameter(Mandatory = $false)]
+        [ValidateScript({$_ | Test-Path -PathType Container})]
+        [String]
+        $TempFolder = $env:TEMP
+    )
+
+    $ErrorActionPreference = "Stop"
+
+    # STEP 1: Connect to the backup store with provided credential
+    Write-Verbose $Strings.ProgressConnectBackupStore -Verbose
+    try
+    {
+        $null = New-PSDrive -Name $BackupStoreDriveName -Root $BackupStorePath -Credential $BackupStoreCredential -PSProvider FileSystem | Out-Null
+    }
+    catch
+    {
+        throw ($Strings.ErrorFailToConnectBackupStore -f $BackupStorePath, $_)
+    }
+
+    Install-Module -Name SqlServer
+    Import-Module SqlServer -DisableNameChecking
+    $sqlCommonParams = @{
+        ServerInstance = $SQLServerInstanceName
+    }
+
+    if (!$SQLCredential)
+    {
+        # STEP 2: Connect to the SQL server with Windows Authentication
+        Write-Verbose ($Strings.ProgressConnectSqlServerDefault -f $SQLServerInstanceName) -Verbose
+        try
+        {
+            $sqlInstance = Get-SqlInstance -ServerInstance $SQLServerInstanceName
+        }
+        catch
+        {
+            throw ($Strings.ErrorFailToConnectSqlServerDefault -f $SQLServerInstanceName, $_)
+        }
+    }
+    else
+    {
+        # STEP 2: Connect to the SQL server with provided credential
+        Write-Verbose ($Strings.ProgressConnectSqlServerProvided -f $SQLServerInstanceName) -Verbose
+        try
+        {
+            $sqlInstance = Get-SqlInstance -ServerInstance $SQLServerInstanceName -Credential $SQLCredential
+            $sqlCommonParams.SQLCredential = $SQLCredential
+        }
+        catch
+        {
+            throw ($Strings.ErrorFailToConnectSqlServerProvided -f $SQLServerInstanceName, $_)
+        }
+    }
+
+    # STEP 3: Create a tmp folder
+    Write-Verbose ($Strings.ProgressCreateTmpFolder -f $SubTmpFolderName, $TempFolder) -Verbose
+    $tmpDir = Join-Path $TempFolder $SubTmpFolderName
+    $null = Remove-Item $tmpDir -Force -Recurse -ErrorAction Ignore | Out-Null
+    $null = New-Item -Type Directory $tmpDir -Force | Out-Null
+    
+    # STEP 4: Get the list of backup files according to the BackupID
+    Write-Verbose ($Strings.ProgressGetBackupSnapshots -f $BackupID) -Verbose
+    $backupRoot = Join-Path $BackupStorePath "MASBackup\progressivebackup"
+    $backupFiles = (Get-ChildItem -Path $backupRoot -Recurse | ? { $_.Name -match $BackupID -and $_.Name -match ".zip" }).FullName
+    $backupRepos = @($BackupRepoNames.Values.GetEnumerator()) | ? { $_ -ne $BackupRepoNames.Subscription }
```

Comment:
Why do we exclude the subscription DB here?

Reply:
The initial thought was to get the snapshots of Subscription repo by Get-RepositoryBackupChain. But you're right, I can actually merge them.